### PR TITLE
Update che template to support latest Che versions (5.22.x and 6.0.0)

### PR DESCRIPTION
**Pull request by l0rd**
This PR is realated to this issue https://github.com/redhat-developer/rh-che/issues/506

- We have changed the Che template to support the latest versions of Che (5.22.1 and 6.0.0)
- We have added some addon variables to be able to configure Che
- We have updated the readme with the instructions to run different versions of Che
- We have set `>=v3.5` as the OpenShift required version

### CODE REVIEW

File: add-ons/che/README.md [hunk 1]
```diff
@@ -56,25 +50,53 @@ we recommand to start Minishift with at least 5GB:
 ### Apply add-on
 If Minishift is already started and che addon is installed. It is possible to deploy che without restarting Minishift:
 
+#### Deploy Che latest (stable)
```
Comment (sunix) [inline add-ons/che/README.md]: maybe specify what is stable (5.22.x ?)

File: add-ons/che/README.md [hunk 2]
```diff
@@ -56,25 +50,53 @@ we recommand to start Minishift with at least 5GB:
...
+
+    $ minishift addons apply --addon-env CHE_HOST_PREFIX="" che
+
+#### Deploy Che nightly (unstable)
```
Comment (sunix) [inline add-ons/che/README.md]: maybe specify what is unstable (che6 ? master)

**Comment by l0rd** [comment]
@sunix thanks for the feedback. I have done some cleanup in the README file.

**Comment by l0rd** [comment]
cc @kameshsampath @LalatenduMohanty for review

**Comment by kameshsampath** [comment]
@l0rd - looks good to me, the `CHE_HOST_PREFIX` as just a carry forward from script based instruction, if it does not matter anymore then we can knock it off.

**Comment by l0rd** [comment]
@kameshsampath I have removed any reference to `CHE_HOST_PREFIX`

**Comment by l0rd** [comment]
@LalatenduMohanty do you think we can merge this? This is currently blocking the release of Che v6.

FYI I have removed the use of `ImageStreams`  because it made it too difficult for a user to deploy it's local che-server image (an important scenario for us).

**Comment by l0rd** [comment]
@gbraad @budhrg @praveenkumar @LalatenduMohanty I know that you have been pretty busy lately but you may be able to merge that now :-)

**Comment by gbraad** [comment]
@minishift/minishift-dev Please take an action on this

**Comment by LalatenduMohanty** [comment]
@l0rd The changes looks fine to me. But there are 4 commits which needs to squashed and mention the issues numbers #73 #74 in the commit message as it will help us to track. Is it ok if I squash the code and merge?

**Comment by gbraad** [comment]
I would suggest you to just go ahead and squash on his behalf.

**Comment by l0rd** [comment]
Yes Lala please go ahead.

**Comment by LalatenduMohanty** [comment]
Merged via https://github.com/minishift/minishift-addons/commit/d13aaa66980127531f6f5fc915ec3a0b6302dec2

**Comment by gbraad** [comment]
👍

**Pull request closed without merging**